src/NzbDrone.Core/Download/MagnetResolverService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using MonoTorrent;
+using MonoTorrent.BEncoding;
+using NLog;
+using NzbDrone.Common.Http;
+
+namespace NzbDrone.Core.Download
+{
+    public interface IResolveMagnetLink
+    {
+        byte[] DownloadTorrentFromMagnet(string magnetUrl);
+    }
+
+    public class MagnetResolverService : IResolveMagnetLink
```

Comment:
I think we are overusing the "service" postfix, something like this could just be called `MagnetLinkResolver`, `service` post fix should be used for entrypoint to main parts of the app, eg, `SeriesService`, `QualityService` etc, not helper components like this,